src/components/RecipeItem.tsx

```diff
@@ -0,0 +1,29 @@
+import {StyledButton, TD, TR} from './styles';
+import Recipe from "./recipe";
+import React from "react";
+
+export default function RecipeItem({
+                                     recipe,
+                                     onEdit,
+                                     onDelete
+                                   }: {
+  recipe: Recipe,
+  onEdit: (id: number) => void,
+  onDelete: (id: number) => void
+}) {
+  return (
+    <TR>
+      <TD>{recipe.name}</TD>
+      <TD>{recipe.description}</TD>
+      <TD>
+        {recipe.ingredients.map(ingredient => (
+          <div key={ingredient.name}>{ingredient.name}</div>
+        ))}
+      </TD>
+      <TD>
+        <StyledButton primary={false} type="button" onClick={() => onEdit(recipe.id || -1)}>Edit</StyledButton>
```

Comment:
what do you think instead of the `|| -1` disable the button when there is no recipe id

Reply:
Much better solution. Thank you!